src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/SimpleHttpClientFactory.cs

```diff
@@ -14,23 +16,66 @@ namespace Microsoft.Identity.Client.PlatformsCommon.Shared
     /// .NET should use the IHttpClientFactory, but MSAL cannot take a dependency on it.
     /// .NET should use SocketHandler, but UseDefaultCredentials doesn't work with it 
     /// </remarks>
-    internal class SimpleHttpClientFactory : IMsalHttpClientFactory
+    internal class SimpleHttpClientFactory : IMsalMtlsHttpClientFactory
     {
         //Please see (https://aka.ms/msal-httpclient-info) for important information regarding the HttpClient.
-        private static readonly Lazy<HttpClient> s_httpClient = new Lazy<HttpClient>(InitializeClient);
+        private static readonly ConcurrentDictionary<string, HttpClient> s_httpClient = new ConcurrentDictionary<string, HttpClient>();
+        private static readonly object s_cacheLock = new object();
 
-        private static HttpClient InitializeClient()
+        private static HttpClient CreateNonMtlsClient()
         {
-            var httpClient = new HttpClient(new HttpClientHandler() { 
-                /* important for IWA */ UseDefaultCredentials = true });
+            var httpClient = new HttpClient(new HttpClientHandler()
+            {
+                /* important for IWA */
+                UseDefaultCredentials = true
+            });
+            HttpClientConfig.ConfigureRequestHeadersAndSize(httpClient);
+
+            return httpClient;
+        }
+
+        private static HttpClient CreateMtlsHttpClient(X509Certificate2 bindingCertificate)
+        {
+            if (s_httpClient.Count > 1000)
```

Comment:
it's better to check size of cache on write, not on read. There are many many more read operations than write operations.